distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Split_Tokenize.enso

```diff
@@ -0,0 +1,246 @@
+tokenize_to_columns table input_column_name pattern case_sensitivity column_count on_problems =
+    expect_text_column table input_column_name <|
+        fan_out_to_columns table input_column_name (ignore_nothing (_.tokenize pattern case_sensitivity)) column_count on_problems
+
+## PRIVATE
+   Tokenizes a column of text into a set of new rows using a regular
+   expression.
+   See `Table.tokenize_to_rows`.
+tokenize_to_rows : Table -> Text | Integer -> Text -> Case_Sensitivity -> Problem_Behavior -> Table
+tokenize_to_rows table input_column_name pattern="." case_sensitivity=Case_Sensitivity.Sensitive on_problems=Report_Error =
+    _ = [on_problems]
+    expect_text_column table input_column_name <|
+        fan_out_to_rows table input_column_name (ignore_nothing (_.tokenize pattern case_sensitivity))
+
+## PRIVATE
+   Transform a table by transforming a column into a set of columns. Takes a
+   function that maps a single element of the input column to a vector of output
+   values. The original column is replaced by the new columns.
+
+   Arguments:
+   - table: The table to transform.
+   - input_column: The column to transform.
+   - function: A function that transforms a single element of `input_column`
+     to multiple values.
+fan_out_to_columns : Table -> Text | Integer -> (Any -> Vector Any) -> Auto | Integer -> Problem_Behavior -> Table | Nothing
+fan_out_to_columns table input_column_name function column_count=Auto on_problems=Report_Error =
+    input_column = table.get input_column_name
+    problem_builder = Problem_Builder.new
+    new_columns_unrenamed = map_columns_to_multiple input_column function column_count problem_builder
+    new_columns = rename_new_columns table new_columns_unrenamed problem_builder
+    new_table = replace_column_with_columns table input_column new_columns
+    problem_builder.attach_problems_after on_problems new_table
+
+## PRIVATE
+   Transform a column by applying the given function to the values in the
+   column. The function produces multiple outputs, so each row is duplicated,
+   with each row getting a distinct output value in place of the original
+   input value. The other column values are just duplicated.
+
+   Arguments:
+   - table: The table to transform.
+   - input_column: The column to transform.
+   - function: A function that transforms a single element of `input_column`
+     to multiple values.
+fan_out_to_rows : Table -> Text | Integer -> (Any -> Vector Any) -> Table
+fan_out_to_rows table input_column_name function =
+    input_column = table.at input_column_name
+    input_storage = input_column.java_column.getStorage
+    num_input_rows = input_storage.size
+
+    # Accumulates the output of the output column values.
+    output_column_builder = make_string_builder 0
+    # Accumulates repeated position indices for the order mask.
+    order_mask_positions = Vector.new_builder 0
+
+    0.up_to num_input_rows . map i->
```

Comment:
```suggestion
    0.up_to num_input_rows . each i->
```

Use `each` instead of `map` if the resulting vector is not used. That way we avoid allocating this result.